src/Reflex/Dom/Contrib/Event.hs

```diff
@@ -0,0 +1,113 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE RankNTypes       #-}
+
+-- | Helper functions for dealing with events
+module Reflex.Dom.Contrib.Event where
+
+import           Control.Applicative      (empty)
+import           Control.Monad            (void)
+import           Control.Monad.IO.Class   (MonadIO)
+import           Data.Time.Clock          (NominalDiffTime)
+import           Reflex
+import qualified Reflex.Dom.Builder.Class as Dom
+import qualified Reflex.Dom.Old           as Dom
+import qualified Reflex.Dom.Widget.Basic  as Dom
+
+eventJoin :: (Reflex t, MonadHold t m) => Event t (Event t a) -> m (Event t a)
+eventJoin = switchHold never
+
+-- | Block those nothing events and only let trough just a's
+noNothing :: FunctorMaybe f => f (Maybe a) -> f a
+noNothing = fmapMaybe id
+
+-- | Do something monadic with an event val
+--   Because of haskell lazyness the things inside a holdevent
+--   don't get evaluated untill the event fires, which makes the first
+--   time slow. However it is good for initialization as we don't
+--   need to load things unused.
+holdEvent ::
+     (Dom.DomBuilder t m, MonadHold t m)
+  => b
+  -> Event t a
+  -> (a -> m b)
+  -> m (Dynamic t b)
+holdEvent val evt fun = Dom.widgetHold (pure val) $ fun <$> evt
+
+-- | Convenience holdEvent for the case where we don't care about the
+--   value.
+holdEvent_ ::
+     (Dom.DomBuilder t m, MonadHold t m) => Event t a -> (a -> m b) -> m ()
+holdEvent_ = fmap void . holdEvent undefined -- we throw away the value
+
+-- | postbuild is imediate for sampling, adding a delay makes it after
+--   widget completes see: https://github.com/reflex-frp/reflex-dom-semui/issues/18
+getReady ::
+     ( PostBuild t m
+     , PerformEvent t m
+     , TriggerEvent t m
+     , MonadIO (Performable m)
+     )
+  => m (Event t ())
+getReady = getPostBuild >>= delay 0
```

Comment:
This isn't actually correct - the delay isn't guaranteed to be long enough that the thing has actually been mounted.  It's true that people tend to use `delay 0` this way (including me), but a correct solution is being worked on [here](https://github.com/reflex-frp/reflex-dom/pull/273).

Reply:
Should I delete it, or change the comment?

Reply:
I think it's probably better to delete it for now.  A function fitting this description would be great, though, when it's possible to implement reliably!